syft/generic/abstract/sendable.py

```diff
@@ -1,11 +1,56 @@
+# from abc import abstractmethod
+from typing import TYPE_CHECKING
+from typing import Union
+
+import syft as sy
 from syft.serde.syft_serializable import SyftSerializable
 from syft.generic.abstract.object import AbstractObject
 
+# from syft.generic.abstract.pointer import AbstractPointer
+from syft.workers.abstract import AbstractWorker
+
+# this if statement avoids circular imports between base.py and pointer.py
+if TYPE_CHECKING:
+    from syft.generic.abstract.pointer import AbstractPointer
+
 
 class AbstractSendable(AbstractObject, SyftSerializable):
     """
     This layers functionality for sending objects between workers on top of AbstractObject.
     """
 
-    def send(self, destination):
-        return self.owner.send_obj(self, destination)
+    def send(
+        self, destination: Union[AbstractWorker, str], garbage_collect_data=None
+    ) -> "AbstractPointer":
+        ptr_id = sy.ID_PROVIDER.pop()
+
+        destination = self.owner.get_worker(destination)
+
+        pointer = self.create_pointer(
+            location=destination,
+            id_at_location=self.id,
+            owner=self.owner,
+            ptr_id=ptr_id,
```

Comment:
I think `ptr_id` can be ignored here, since it will be assigned a value using `sy.ID_PROVIDER` when `AbstractPointer` is initialized with `id=None`.

Reply:
That's true in the concrete case, but can't be depended on in this abstract class. (If the id assignment moved into this class, then I'd be on board with ignoring the pointer ids here.)

Reply:
I am not sure to understand the point. I meant if we are going always to use 

```
id = sy.ID_PROVIDER.pop() 
```

Inside the above `send()` method then it becomes exactly equivalent to ignoring the attribute altogether (I mean only inside the `send()` method)

Reply:
`send` isn’t necessarily the only place where `create_pointer` will get called. If `create_pointer` was an implementation detail of `send` or if it was always responsible for assigning ids (absorbing that responsibility from other classes), then we could ignore the param.

Reply:
This makes sense